BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/SelfDiagnostics/SelfDiagnosticsEventListener.cs

```diff
@@ -71,6 +71,11 @@ public override void Dispose()
         /// <returns>The position of the buffer after the last byte of the resulting sequence.</returns>
         internal static int EncodeInBuffer(string str, bool isParameter, byte[] buffer, int position)
         {
+            if (str == null)
```

Comment:
It's better to use `String.IsNullOrEmpty`.

Reply:
Fixed.